### PR TITLE
Add cross application tracing using newrelic

### PR DESCRIPTION
**Pull request by mouadino**
- Get rid of monkey patching by making needed component extendable (observer pattern)
- Add support for newrelic cross application tracing replacing normal function tracing for rpc
- Add new tracing package that allow pluggeable tracing scenarios to allow developer to instrument
their services (for now only newrelic is supported)

### CODE REVIEW

File: lymph/core/channels.py [hunk 1]
```diff
@@ -35,6 +37,13 @@ def get(self, timeout=1):
     def close(self):
         del self.server.channels[self.request.id]
 
+    def on_new_message(self, func):
+        self._callbacks.append(func)
+
+    def _tigger_callbacks(self, msg):
```
Comment (manisenkov) [inline lymph/core/channels.py]: typo

**Comment by emulbreh** [comment]
lymph.tracing seems to be unused and undocumented.

**Comment by mouadino** [comment]
Unused that's normal since it's new and it meant to be used by service writers :)
Undocumented b/c I would like to see how it will get used first in some of our service before publishing the API, since changing a published API in lymph is very hard.

Basically yes, for now only ``trace_context`` is supplied, and what service writer have to do is pass the tracer to the component that they would like to trace in depth, and use it there.

**Comment by mouadino** [comment]
Putting this in hold since I don't want to waste more time on this.

**Pull request closed without merging**